### DISCUSSION TITLE
getServerSideProps and ssrExchange invalidation

### DISCUSSION THREAD

**Discussion by danielr18** [urql-graphql/urql#2686] [Q&A]
Shouldn't data coming from `getServerSideProps` not be ignored when calling `restoreData` regardless of it being invalidated before?

**Comment by kitten** [comment]
Yes, as long as your setup contains an `ssrExchange` in front of a `fetchExchange` then the former will synchronously yield its cached results and stand-in for results that'd otherwise be fetched 

In other words, it emits results that the server-side has already seen just once. Afterwards, the cache exchange you've set up takes over

**Reply by danielr18** [reply]
Here's an example, we have the following page:

/time: it queries the current timestamp in getServerSideProps

The timestamp query requestPolicy is `network-only`, and I would expect the timestamp to always come from ssrExchange.

However the ssrExchange.restoreDara method doesn't restore operations that has been previously invalidated.

Wouldn't this cause the second time /time page is visited to hit  the fetchExchange because the operation is not cached due to the requestPolicy and it was invalidated in the ssrExchange the  first time the page was visited?
